### PR TITLE
VxScan: Stream CVR Exports

### PR DESCRIPTION
**Pull request by adghayes**
## Overview
Does two things:
1. Makes CVR exports streaming (from the backend to the frontend to the download) to accommodate larger CVRs
2. Sets up option to get CVRs without inline images for tally purposes, to speed up polls closing

Now that we are downloading the file from the backend, we should set it's file name via the content disposition. However, currently we don't have access to the machine ID (which needs to be in the filename) in the backend. That's easy to do, but to do it right, we should abstract out the machine config provider to `utils` which I didn't want to bother with on the fly.

## Testing Plan 
- Edited a bunch of tests, added a few
- Manual testing _in kiosk-brower_
- Note: I have not tested this in the browser. We have some logic for downloads in the browser for dev purposes, and I have not validated those, although they should be intact

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: frontends/precinct-scanner/src/utils/save_cvr_export_to_usb.test.ts
Comment (adghayes) [inline frontends/precinct-scanner/src/utils/save_cvr_export_to_usb.test.ts:86]: the default "Save As" path for CVRs is no longer within a specific directory, but just at the root of the USB drive. seems unimportant

File: services/scan/src/precinct_scanner_app.ts
Comment (adghayes) [inline services/scan/src/precinct_scanner_app.ts:467]: this is really a placeholder, and could be removed, because the frontend is overriding the filename in the content-disposition

Comment (adghayes) [reply services/scan/src/precinct_scanner_app.ts:467]: maybe it's bad to have it in for now because if the `TODO` isn't done and and the endpoint is relied on later it could give CVR filenames without correct machine ID

File: services/scan/src/store.test.ts
Comment (adghayes) [inline services/scan/src/store.test.ts:722]: test which is, for the most part, shamelessly copied

File: services/scan/src/store.ts
Comment (adghayes) [inline services/scan/src/store.ts:953]: didn't end up being important after I landed on using the `Passthrough` stream, but still makes sense to close the stream

File: frontends/precinct-scanner/src/api/scan.test.ts
Comment (eventualbuddha) [inline frontends/precinct-scanner/src/api/scan.test.ts:30]: This doesn't appear to actually validate anything about the request.

File: frontends/precinct-scanner/src/utils/save_cvr_export_to_usb.ts
Comment (eventualbuddha) [inline frontends/precinct-scanner/src/utils/save_cvr_export_to_usb.ts:75]:
```suggestion
        errorMessage = `Unable to get CVR file: ${error.kind} (status=${error.response.statusText})`;
```

File: frontends/precinct-scanner/src/api/scan.test.ts
Comment (adghayes) [reply frontends/precinct-scanner/src/api/scan.test.ts:30]: Wouldn't the test fail if `fetchMock` was not matched?

File: libs/api/src/services/scan/index.ts
Comment (eventualbuddha) [inline libs/api/src/services/scan/index.ts:512]: Maybe this?

```ts
export type ExportRequest = Optional<{ skipImages: boolean }>;
```

File: frontends/precinct-scanner/src/api/scan.test.ts
Comment (adghayes) [reply frontends/precinct-scanner/src/api/scan.test.ts:30]: `fetchMock` expecting a request meeting a certain filter was, in my mind, what was being validated.

File: services/scan/src/precinct_scanner_app.test.ts
Comment (eventualbuddha) [inline services/scan/src/precinct_scanner_app.test.ts:1407]: It's a little hard to tell what the `true` here refers to. Maybe make it an options object instead of just a bare argument?

File: libs/api/src/services/scan/index.ts
Comment (adghayes) [reply libs/api/src/services/scan/index.ts]: Done.

File: services/scan/src/precinct_scanner_app.ts
Comment (eventualbuddha) [inline services/scan/src/precinct_scanner_app.ts:472]: Why not `store.exportCvrs(response, skipImages)`? Why do we have the `PassThrough` stream?

Comment (adghayes) [reply services/scan/src/precinct_scanner_app.ts:472]: `exportCvrs` is expecting a Writable interface, but we also need it to be a Readable interface to pipe it into the response. The `Passthrough` just gives us both. Do you know an easier way?

File: frontends/precinct-scanner/src/api/scan.test.ts
Comment (eventualbuddha) [reply frontends/precinct-scanner/src/api/scan.test.ts:30]: Ah, you're right. The `body` of the first argument does match the request body.

Comment (adghayes) [reply frontends/precinct-scanner/src/api/scan.test.ts:30]: Do you think expecting `fetchMock.done()` would make it any more clear?

File: services/scan/src/precinct_scanner_app.test.ts
Comment (adghayes) [reply services/scan/src/precinct_scanner_app.test.ts:1407]: Will do.

File: frontends/precinct-scanner/src/api/scan.test.ts
Comment (eventualbuddha) [reply frontends/precinct-scanner/src/api/scan.test.ts:30]: Nah, we do this "just call a thing and if it doesn't blow up it passes" elsewhere.

File: services/scan/src/precinct_scanner_app.ts
Comment (eventualbuddha) [inline services/scan/src/precinct_scanner_app.ts:474]: Should we make this change with the central scanner app as well?

Comment (adghayes) [reply services/scan/src/precinct_scanner_app.ts:474]: Yeah, definitely, but I think we can punt that for NH because NH would be using the central scanner. Is that right @mattroe?

Comment (eventualbuddha) [reply services/scan/src/precinct_scanner_app.ts:474]: I believe that's the case. I just thought if it's easy let's do it now. Your call.

Comment (adghayes) [reply services/scan/src/precinct_scanner_app.ts:474]: https://github.com/votingworks/vxsuite/issues/2682

File: services/scan/src/precinct_scanner_app.ts
Comment (eventualbuddha) [reply services/scan/src/precinct_scanner_app.ts:472]: The `response` is writable. I just changed it to this and it appears to work with kiosk-browser and pass the tests.

File: services/scan/src/precinct_scanner_app.ts
Comment (adghayes) [reply services/scan/src/precinct_scanner_app.ts:474]: @carolinemodic do you want to merge this in imminently? In which case I'll put off the change and testing for central scanner.

Comment (carolinemodic) [reply services/scan/src/precinct_scanner_app.ts:474]: Yeah lets do the central scanner in a separate PR as thats not used for NH

File: services/scan/src/precinct_scanner_app.ts
Comment (adghayes) [reply services/scan/src/precinct_scanner_app.ts:472]: Oh, that makes a lot of sense. Will change to be so.

**Review by eventualbuddha — COMMENTED** [review]
As a side effect of these changes we seem to have lost the ability to download the file in Chrome. I suggest we fix that or completely remove that as an option.

File: services/scan/src/store.ts
Comment (eventualbuddha) [inline services/scan/src/store.ts:956]: One thing to note is that we're not really using the writable stream API correctly. This can lead to excessive buffering and potentially a much slower export operation due to the memory usage/GC/swapping going on. See https://enlear.academy/nodejs-backpressuring-in-streams-52638f505e1b for what we should be doing differently, specifically waiting for a `drain` event after each `write` that returns `false`.

**Comment by adghayes** [comment]
Working on fixing the Chrome download.

File: services/scan/src/store.ts
Comment (adghayes) [reply services/scan/src/store.ts:956]: Interesting. I'm not sure exactly how it will look - I tried just waiting on the writeStream to drain but we can't do that because we have an ongoing SQL query that we're iterating through, so may have to change the way we're working through the data. Not going to touch that for now.

**Review by eventualbuddha — APPROVED** [review]
LGTM once the Chrome download is fixed or disabled.

File: frontends/precinct-scanner/src/components/export_results_modal.tsx
Comment (adghayes) [inline frontends/precinct-scanner/src/components/export_results_modal.tsx:170]: underlying feature is disabled but also disabling the button

File: services/scan/src/precinct_scanner_app.ts
Comment (adghayes) [inline services/scan/src/precinct_scanner_app.ts:463]: Just a clearer non-id if anyone ever ends up using this.

File: services/scan/src/precinct_scanner_app.ts
Comment (adghayes) [reply services/scan/src/precinct_scanner_app.ts:472]: Had to use Passthrough here, otherwise it didn't work!

File: services/scan/src/store.ts
Comment (adghayes) [inline services/scan/src/store.ts:953]: I don't know if this does anything but it seems better to have fewer rather than more write events.

**Comment by adghayes** [comment]
There are two issues with the way this PR that we will eventually have to circle back to:
1. Downloading CVRs via Chrome no longer works. This can be enabled by changing the endpoint from`POST` to `GET`
2. For some reason, the CVR export doesn't return headers until it's done collecting data, even if we `flushHeaders` this may have to do with the ways in which we're `pipe`ing into the response. For now, it means that the "Custom" location export works it doesn't prompt for the location until after the file is complete.

**Pull request merged**
